balrog/policy.py

```diff
@@ -51,9 +67,14 @@ def check(self, permission, *args, **kwargs):
 
         :param permission: Permission name.
         :return: `True` if identity role has this permission.
+        :raises: `PermissionNotFound` when no permission is found.
         """
+        self._check_permission(permission)
         identity = self.get_identity(*args, **kwargs)
-        role = self.get_role(identity, *args, **kwargs)
+        try:
+            role = self.get_role(identity, *args, **kwargs)
+        except exceptions.RoleNotFound:
```

Comment:
we loose the check for invalid roles, but we add a check for invalid persmissions, well, in total it's 0 :)